dask_bigquery/core.py

```diff
@@ -109,13 +125,35 @@ def read_gbq(
       list of columns to load from the table
     read_kwargs: dict
       kwargs to pass to read_rows()
+    fwd_creds: bool
+      Set to True if user desires to forward credentials to the workers. Default to False.
 
     Returns
     -------
         Dask DataFrame
     """
     read_kwargs = read_kwargs or {}
-    with bigquery_clients(project_id) as (bq_client, bqs_client):
+
+    if fwd_creds:
+        creds_path = os.environ.get("GOOGLE_APPLICATION_CREDENTIALS")
+        if creds_path is None:
+            raise ValueError("No credentials found")
+
+        credentials = service_account.Credentials.from_service_account_file(
```

Comment:
You might want to call

https://googleapis.dev/python/google-auth/latest/reference/google.auth.html#google.auth.default

instead. That'll find the `GOOGLE_APPLICATIONS_CREDENTIALS` environment variable if available, but also allow for other authentication methods such as `gcloud auth application-default login`.

Reply:
I believe any kind of credentials it returns should have a `refresh` function, just like the service account credentials.

Reply:
Thanks Tim, I will take a look at this.